sensible default types

tommyknows opened the pull request:
**fix: reference to scan config**

This commit fixes the Helm chart's rendering of our scan types in the
ClusterRole.



**feat: improve default types**

This commit updates / adjusts the default types that are being scanned.
This list has been built based on the kubernetes-monitor RBAC list (see
[TIKI-39](https://snyksec.atlassian.net/browse/TIKI-39) for a link).


[TIKI-39]: https://snyksec.atlassian.net/browse/TIKI-39?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Commit:
fix: reference to scan config

This commit fixes the Helm chart's rendering of our scan types in the
ClusterRole.

Commit:
feat: improve default types

This commit updates / adjusts the default types that are being scanned.
This list has been built based on the kubernetes-monitor RBAC list (see
[TIKI-39](https://snyksec.atlassian.net/browse/TIKI-39) for a link).

craigfurman reviewed (approved):
LGTM! If you haven't already, it might be worth pointing a local scanner at dev16's kube API through teleport just to check this works as expected.

Commit:
fix: return NotFoundError for not found groups

So far we didn't return a "not found" error, so the controller didn't
start when we provided a group that didn't exist. This commit fixes
that, which means that we're now skipping inexistent groups (like if a
CRD isn't installed).

Commit:
feat: improve APIVersion selection

This commit improves the APIVersion selection for cases where the user
has asked us to scan for the preferred version, but not all resource
types might exist at that version. So far, we've then skipped all
resources that are not present in the preferredVersion.

With these changes, we'll fall back to searching all other versions as
well.

craigfurman reviewed (approved):
Great use of separate commits and messages to help the reviewer, and https://github.com/snyk/kubernetes-scanner/pull/16/commits/413d93df293249bced33009e7564f3a1193ea526 in particular is above and beyond 🥇
